invoices/invoiceregistry.go

```diff
@@ -57,6 +57,10 @@ type RegistryConfig struct {
 	// send payments.
 	AcceptKeySend bool
 
+	// GcCanceledInvoices if set, we'll attempt to garbage collect all
+	// canceled invoices upon start.
+	GcCanceledInvoices bool
```

Comment:
it might be useful to distinguish delete-on-startup from delete-on-the-fly

Reply:
done (on the fly part is in a separate commit)